### PR TITLE
Fixes for multi select Case list

### PR DESCRIPTION
**Pull request by shubham1g5**
Some minor fixes that came up during testing. 

https://github.com/dimagi/commcare-core/pull/1097

### CODE REVIEW

File: src/test/java/org/commcare/formplayer/tests/MultiSelectCaseListTest.java [hunk 1]
```diff
@@ -65,6 +68,19 @@ public void testNormalMultiSelectCaseList() throws Exception {
...
+    @Test
+    public void testMultiSelectCaseListWithPreValidatedInput() throws Exception {
+        Mockito.doReturn(true)
+                .when(restoreFactoryMock).isConfirmedSelection(any());
+        String[] selections = new String[]{"0", "1", "use_selected_values"};
+        String[] selectedValues =
+                new String[]{"5e421eb8bf414e03b4871195b869d894", "3512eb7c-7a58-4a95-beda-205eb0d7f163"};
+        NewFormResponse formResp = sessionNavigateWithSelectedValues(selections, APP, selectedValues,
+                NewFormResponse.class);
+        sessionNavigateWithSelectedValues(selections, APP, selectedValues,
```
Comment (snopoke) [inline src/test/java/org/commcare/formplayer/tests/MultiSelectCaseListTest.java:80]: how did this fail before? It's not clear from the test what's actually being tested.

File: src/main/java/org/commcare/formplayer/services/RestoreFactory.java [hunk 2]
```diff
@@ -178,6 +183,9 @@ public void configure(AuthenticatedRequestBean authenticatedRequestBean, HqAuth
         log.info(String.format("configuring RestoreFactory from authed request with arguments " +
                         "username = %s, asUsername = %s, domain = %s",
                 username, asUsername, domain));
+        if (authenticatedRequestBean instanceof InstallRequestBean) {
+            storageFactory.configure(((InstallRequestBean)authenticatedRequestBean));
```
Comment (snopoke) [inline src/main/java/org/commcare/formplayer/services/RestoreFactory.java:187]: this is already happening in the [AppInstallAspect](https://github.com/dimagi/formplayer/blob/dffa67cb818c8474eed05ec6983c8a0ad7483198/src/main/java/org/commcare/formplayer/aspects/AppInstallAspect.java#L43). I think the best approach for now is to add the `@AppInstall` annotation to whatever view it is missing from.

File: src/test/java/org/commcare/formplayer/tests/MultiSelectCaseListTest.java (hunk 1 above)
Comment (shubham1g5) [reply src/test/java/org/commcare/formplayer/tests/MultiSelectCaseListTest.java]: it was hitting a session navigation exception due to the datum being set as  `use_selected_values` [here](https://github.com/dimagi/formplayer/blob/fixesForMultiSelect/src/main/java/org/commcare/formplayer/session/MenuSession.java#L176) . Probably [this](https://github.com/dimagi/formplayer/pull/1122/commits/4b585d8cb76a493b00f23e00cdf37b6f7a52856e) is better failure mode.

File: src/test/resources/archives/case_claim_with_multi_select/suite.xml [hunk 3]
```diff
@@ -232,7 +232,7 @@
     <command id="m1-f1"/>
   </menu>
   <remote-request>
-    <post url="http://localhost:8000/a/test/phone/claim-case/" relevant="true()">
+    <post url="http://localhost:8000/a/test/phone/claim-case/" relevant="$case_id != ''">
```
Comment (snopoke) [inline src/test/resources/archives/case_claim_with_multi_select/suite.xml:235]: Does `$case_id` refer to the value of the data element? Where is that set in the eval context?

Comment (snopoke) [reply src/test/resources/archives/case_claim_with_multi_select/suite.xml:235]: Ah: https://github.com/dimagi/commcare-core/pull/1097

**Review by snopoke — APPROVED** [review]

**Pull request merged**